pkg/api/v1/orchestrator/routes/taskKV.go

```diff
@@ -199,6 +199,16 @@ func (t *taskKVImpl) publish(
 	if onlyTimestamp {
 		currTask.UpdatedAt = time.Now()
 	} else {
+		// clear stale task entry
+		if rctypes.StateIsComplete(currTask.State) ||
+			(time.Since(currTask.UpdatedAt) > rctypes.StaleThreshold && currTask.ID != task.ID) {
+			if err := t.kv.Delete(key); err != nil {
```

Comment:
This `Delete` is not required, you could simply update with the latest task entry.

Reply:
I'd like to take this up in a separate PR,

For whatever reason, I could not get it to work with a `put` call - `nats: wrong last sequence: 5: key exists` 
```
diff --git a/pkg/api/v1/orchestrator/routes/taskKV.go b/pkg/api/v1/orchestrator/routes/taskKV.go
index 1a31af8..bfb9f97 100644
--- a/pkg/api/v1/orchestrator/routes/taskKV.go
+++ b/pkg/api/v1/orchestrator/routes/taskKV.go
@@ -199,10 +199,16 @@ func (t *taskKVImpl) publish(
        if onlyTimestamp {
                currTask.UpdatedAt = time.Now()
        } else {
-               // clear stale task entry
+               // overwrite stale task
                if rctypes.StateIsComplete(currTask.State) ||
                        (time.Since(currTask.UpdatedAt) > rctypes.StaleThreshold && currTask.ID != task.ID) {
-                       if err := t.kv.Delete(key); err != nil {
+                       task.CreatedAt = time.Now()
+                       taskJSON, errMarshal := task.Marshal()
+                       if errMarshal != nil {
+                               return failed(errMarshal)
+                       }
+
+                       if _, err := t.kv.Put(key, taskJSON); err != nil {
                                return errors.Wrap(errPublishTask, err.Error())
                        }
```